### PR TITLE
Misc fixes, pt. 4

### PR DESCRIPTION
**Pull request by bugaevc**
cc @alimpfard for the waiting fix

### CODE REVIEW

File: Services/WindowServer/WindowManager.cpp [hunk 1]
```diff
@@ -761,10 +761,8 @@ void WindowManager::process_mouse_event(MouseEvent& event, Window*& hovered_wind
     HashTable<Window*> windows_who_received_mouse_event_due_to_cursor_tracking;
 
     for (auto* window = m_windows_in_order.tail(); window; window = window->prev()) {
-        if (!window->global_cursor_tracking())
+        if (!window->global_cursor_tracking() || !window->is_visible() || window->is_minimized())
             continue;
-        ASSERT(window->is_visible());    // Maybe this should be supported? Idk. Let's catch it and think about it later.
-        ASSERT(!window->is_minimized()); // Maybe this should also be supported? Idk.
```
Comment (awesomekling) [inline Services/WindowServer/WindowManager.cpp:767]: I love the confidence in these comments

**Comment by alimpfard** [comment]
@bugaevc 
How would this work for `waitpid(-1, WNOHANG)`?

**Comment by alimpfard** [comment]
Would we still get _a_ child pid?

**Comment by bugaevc** [comment]
This does work for `waitpid(-1, WNOHANG)`, and we do get the correct `child_pid`, at least in case there was somebody to wait for (see SystemServer). If it returns because there's nobody to wait for, it will *probably* result in `-ECHILD` instead of the proper zero return code & zero siginfo. That's something further to investigate & fix, if you want to work on that.

**Comment by alimpfard** [comment]
Alright.
I'll that to my TODO list, but I don't think it's a pressing issue.

**Pull request merged**